src/conn/conn_api.c

```diff
@@ -2740,7 +2740,14 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler, const char *c
         WT_ERR(wt_session->salvage(wt_session, WT_METAFILE_URI, NULL));
     }
 
-    /* Initialize the connection's base write generation. */
+    /*
+     * Initialize the connection's base write generation.
+     *
+     * We'll write over this value after performing rollback to stable however, we need to set it
+     * here. The logic below will involve opening up the metadata file and if the connection-wide
+     * base write generation is uninitialized, we'll tag the btree with the wrong base write gen and
+     * incorrectly interpret transaction ids during rollback to stable.
+     */
     WT_ERR(__wt_metadata_init_base_write_gen(session));
```

Comment:
@tetsuo-cpp Updating the connection base writegen internally updates all the btree base writegen when the btree is opened. This change reset the transaction ids when the pages get loaded into cache? This will give a problem to the RTS to find out which transactions need to rollback.

Reply:
Ok, I had a chat with @kommiharibabu and I learned that we specifically want to see transaction ids during rollback to stable in the future.

These failures we were seeing before are a result of NOT wiping the transaction ids. We see high transaction ids from the previous run and when we try to write checkpoint information in metadata, we think we've conflicted with a concurrent operation.

This problem exists regardless of #6112. The only reason we are seeing this failure now is that we were not actually wiping the ids during rollback to stable before.

The initialisation used to look like:
```
btree->write_gen = btree->base_write_gen = WT_MAX(ckpt->write_gen + 1, conn->base_write_gen);
```

So while it would not initialize to be equal to `conn->base_write_gen`, it would continue where it left off and ids would still get wiped.

I added logging to 2335963 (before we merged in `develop`) and I verified that base write generations were still getting initialized and if I added some logic to set the `btree->base_write_gen` to 0 in the case of the metadata file, I saw the same issue.

@raviprakashgiri29 @kommiharibabu 
I think the most reasonable thing to do is remove c815dbc when you change rollback to stable to use transaction ids. At that point, you can handle the metadata issue (maybe wipe the transaction ids for metadata file?).